src/main/java/com/ita/if103java/ims/service/impl/LoginServiceImpl.java

```diff
@@ -0,0 +1,87 @@
+package com.ita.if103java.ims.service.impl;
+
+import com.ita.if103java.ims.dao.UserDao;
+import com.ita.if103java.ims.dto.UserLoginDto;
+import com.ita.if103java.ims.entity.User;
+import com.ita.if103java.ims.mapper.UserDtoMapper;
+import com.ita.if103java.ims.security.JwtTokenProvider;
+import com.ita.if103java.ims.service.LoginService;
+import com.ita.if103java.ims.service.MailService;
+import com.mysql.cj.exceptions.PasswordExpiredException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.UUID;
+
+import static com.ita.if103java.ims.config.MailMessagesConfig.FOOTER;
+import static com.ita.if103java.ims.config.MailMessagesConfig.RESET_PASSWORD;
+import static com.ita.if103java.ims.util.TokenUtil.isValidToken;
+
+@Service
+public class LoginServiceImpl implements LoginService {
+
+    @Value("${mail.resetPasswordURL}")
+    private String resetPasswordURL;
+
+    private UserDao userDao;
+    private PasswordEncoder passwordEncoder;
+    private JwtTokenProvider jwtTokenProvider;
+    private MailService mailService;
+    private UserDtoMapper mapper;
+
+    @Autowired
+    public LoginServiceImpl(UserDao userDao, PasswordEncoder passwordEncoder, JwtTokenProvider jwtTokenProvider,
+                            MailService mailService, UserDtoMapper mapper) {
+        this.userDao = userDao;
+        this.passwordEncoder = passwordEncoder;
+        this.jwtTokenProvider = jwtTokenProvider;
+        this.mailService = mailService;
+        this.mapper = mapper;
+    }
+
+    @Override
+    public String signIn(UserLoginDto userLoginDto) {
+        User user = userDao.findByEmail(userLoginDto.getUsername());
+        if (passwordEncoder.matches(userLoginDto.getPassword(), user.getPassword())) {
+            return jwtTokenProvider.createToken(userLoginDto.getUsername(),
+                userDao.findByEmail(userLoginDto.getUsername()).getRole());
+        } else {
+            throw new IllegalArgumentException("Password isn't correct");
```

Comment:
From security point of view, better to throw UserOrPasswordIsIncorectException with message "Username or password is incorrect"